src/OpenLoco/TownManager.cpp

```diff
@@ -59,6 +63,58 @@ namespace OpenLoco::TownManager
 
     static auto& rawTowns() { return getGameState().towns; }
 
+    // 0x00497348
+    void sub_497348()
+    {
+        for (auto& town : towns())
+        {
+            town.var_38 = 0;
+            town.population = 0;
+            town.var_34 = 0;
+            memset(town.pad_150, 0, sizeof(town.pad_150));
+        }
+
+        Map::TilePosRangeView tileLoop{ { 1, 1 }, { Map::map_columns - 1, Map::map_rows - 1 } };
+        for (const auto& tilePos : tileLoop)
+        {
+            auto tile = Map::TileManager::get(tilePos);
+            auto* building = tile.building();
```

Comment:
As discussed in previous comment we should be checking every building on the tile not just the one.